### PR TITLE
Code documentation

### PR DESCRIPTION
**Pull request by carcos-wolox**
## Summary ##
- Separate cards and simple animations.
- Add documentation for animation functions.

## Screenshot ##
Screen Shot 2019-05-07 at 4 23 52 PM

## Screencast ##
documentation

## Known issues ##

Most of the methods in Animations are examples of how to use AnimationTypes and normal animatios, so the documentation just explain then in a general way

### CODE REVIEW

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 84.3
:white_check_mark: Code Quality (Swiftlint): 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/code-documentation) for more details.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 84.3
:white_check_mark: Code Quality (Swiftlint): 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/code-documentation) for more details.

File: WolmoCore/Extensions/Animations/MixedAnimation.swift [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import Foundation
 import UIKit
 
+/* Concatenate diffente affine transformations in one single transformation */
```
Comment (acid-wolox) [inline WolmoCore/Extensions/Animations/MixedAnimation.swift]: `diffente affine` ? Maybe a typo?

File: AnimationDemo/CardsViewController.swift [hunk 2]
```diff
@@ -1,96 +1,115 @@
...
         let swapViewsLimit = halfWidthScreen + 75
-
+        
         switch gesture.state {
         case .began:
             rotationAnimator = UIViewPropertyAnimator(duration: 4, curve: UIViewAnimationCurve.easeInOut) {
                 target.transform = CGAffineTransform(rotationAngle: (CGFloat.pi * 15.0) / 180.0)
             }
-
+            print((CGFloat.pi * 15.0) / 180.0)
```
Comment (acid-wolox) [inline AnimationDemo/CardsViewController.swift]: 🤔

File: AnimationDemo/CardsViewController.swift [hunk 3]
```diff
@@ -99,81 +118,64 @@ extension ViewController {
...
-        //            view.alpha = 1
-        //        }
-
         view
             .mixedAnimation(withDuration: 0.25)
             .action(positionX: halfWidthScreen, positionY: view.center.y)
             .transform(scaleX: 1, scaleY: 1)
-            .action(alpha: 1)
             .action(moveTo: .front)
+            .action(alpha: 1)
```
Comment (acid-wolox) [inline AnimationDemo/CardsViewController.swift]: Why has this line changed its position?

File: AnimationDemo/AnimationsViewController.swift [hunk 4]
```diff
@@ -217,71 +293,61 @@ extension ViewController {
...
+        
         return (coreAnimation: animation, normalAnimation: .none)
     }
-
-    private func setupTransformRotateAndActionScaleAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates a MixedAnimation that rotates and scales the view
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 5]
```diff
@@ -217,71 +293,61 @@ extension ViewController {
...
+        
         return (coreAnimation: animation, normalAnimation: .none)
     }
-
-    private func setupTransformScaleAndActionTranslateAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates four MixedAnimations and sets them in a single ChainedAnimation that executes one after the previous one is completed, repeating them
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 6]
```diff
@@ -190,23 +252,37 @@ extension ViewController {
...
+        
         return (coreAnimation: animation, normalAnimation: .none)
     }
-
-    private func setupTransformTranslateAndScaleMixedAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates four MixedAnimations and sets them in a single ChainedAnimation that executes one after the previous one is completed, repeating them
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 7]
```diff
@@ -190,23 +252,37 @@ extension ViewController {
...
+        
         return (coreAnimation: animation, normalAnimation: normalAnimation)
     }
-
-    private func setupTransformTranslateAndScaleSimpleAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates a SimpleAnimation that adds some transformation to the view
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 8]
```diff
@@ -156,31 +204,45 @@ extension ViewController {
...
+        
         return (coreAnimation: animation, normalAnimation: .none)
     }
-
-    private func setupTransformRotateAndScaleAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates a MixedAnimation that scals the view and and a normal animation
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 9]
```diff
@@ -156,31 +204,45 @@ extension ViewController {
...
+        
         return (coreAnimation: animation, normalAnimation: .none)
     }
-
-    private func setupChainedDifferentTypesAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates a SimpleAnimation and a MixedAnimation and sets them in a single chained animation that executes one after the previous one is completed, without repeating them
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 10]
```diff
@@ -130,19 +170,27 @@ extension ViewController {
...
+        
         return (coreAnimation: .none, normalAnimation: animation)
     }
-
-    private func setupChainedAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates 2 different mixed animations and set them in a single chained animation that executes one after the previous one is completed, and repeats them in a loop
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

File: AnimationDemo/AnimationsViewController.swift [hunk 11]
```diff
@@ -99,26 +91,74 @@ extension ViewController {
...
         return (coreAnimation: animation, normalAnimation: .none)
     }
-
-    //swiftlint:disable:next function_body_length
-    private func setupChainAnimation(_ view: UIView) -> (coreAnimation: AnimationType?, normalAnimation: (() -> Void)?) {
+    
+    /**
+     Creates a normal animation of a chain of transformations, adding each one after the previous one is completed
+     
+     - Parameter view: view to be animates
```
Comment (acid-wolox) [inline AnimationDemo/AnimationsViewController.swift]: *animated

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 84.3
:white_check_mark: Code Quality (Swiftlint): 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/code-documentation) for more details.

**Pull request merged**